lib/pbench/test/unit/server/test_user_management_cli.py

```diff
@@ -0,0 +1,251 @@
+                    TestUserManagement.PSWD_SWITCH,
+                    TestUserManagement.PSWD_TEXT,
+                    TestUserManagement.EMAIL_SWITCH,
+                    TestUserManagement.EMAIL_TEXT,
+                    TestUserManagement.FIRST_NAME_SWITCH,
+                    TestUserManagement.FIRST_NAME_TEXT,
+                    TestUserManagement.LAST_NAME_SWITCH,
+                    TestUserManagement.LAST_NAME_TEXT,
+                    TestUserManagement.ROLE_SWITCH,
+                    "ADMN",
+                ],
+            )
+            assert result.exit_code == 2
+            assert (
+                result.stderr_bytes
+                == b"Usage: user-create [OPTIONS]\nTry 'user-create --help' for help.\n"
+                + b"\nError: Invalid value for '--role': invalid choice: ADMN. (choose from ADMIN)\n"
+            )
+
+    @staticmethod
+    @responses.activate
+    def test_valid_user_delete(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            result = TestUserManagement.register_user(server_config, pytestconfig)
+            assert result.exit_code == 0, result.stderr
+
+            result = runner.invoke(user_delete, args=[TestUserManagement.USER_TEXT])
+            assert result.exit_code == 0, result.stderr
+            assert result.stdout == "User test_user deleted\n"
+
+    @staticmethod
+    @responses.activate
+    def test_invalid_user_delete(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            # Give username that doesn't exists to delete
+            result = runner.invoke(user_delete, args=["wrong_username"])
+            assert result.exit_code == 1
+            assert result.stderr == "User wrong_username does not exist\n"
+
+    @staticmethod
+    @responses.activate
+    def test_valid_user_list(client, server_config, pytestconfig):
+        with client:
+            runner = CliRunner(mix_stderr=False)
+            result = TestUserManagement.register_user(server_config, pytestconfig)
+            assert result.exit_code == 0, result.stderr
+
+            result = runner.invoke(user_list,)
+            assert result.exit_code == 0, result.stderr
+            assert (
+                result.stdout
+                == USER_LIST_HEADER_ROW
+                + "\n"
+                + USER_LIST_ROW_FORMAT.format(
+                    TestUserManagement.USER_TEXT,
+                    TestUserManagement.FIRST_NAME_TEXT,
+                    TestUserManagement.LAST_NAME_TEXT,
+                    datetime.datetime.now().strftime("%Y-%m-%d"),
+                    TestUserManagement.EMAIL_TEXT,
+                )
+                + "\n"
+            )
+
+    @staticmethod
+    @responses.activate
+    @pytest.mark.parametrize(
+        "email, role, firstname, lastname, username",
+        [("new_test@domain.com", "ADMIN", "newfirst", "newlast", "newuser")],
```

Comment:
That's interesting that the quoted list appears to work here as parameter names, but a dict might be easier to read with a list of parameter sets. Also, you're only specifying a single set of values, so the `parametrize` decoration seems pointless here anyway.

Reply:
Apparently, a quoted list is a [documented option](https://docs.pytest.org/en/stable/reference.html#pytest.python.Metafunc.parametrize), as are `list` and `tuple`.

But, yes, Nihkil, you have this set up with only a single testcase.  I think we were envisioning something more like this:
```
    @pytest.mark.parametrize(
        "switch, value",
        [(TestUserManagement.USER_SWITCH, "new_test"),
         (TestUserManagement.EMAIL_SWITCH, "new_test@domain.com"),
         (TestUserManagement.ROLE_SWITCH, "ADMIN"),
         (TestUserManagement.FIRST_NAME_SWITCH, "newfirst"),
         (TestUserManagement.LAST_NAME_SWITCH, "newlast"),
         (TestUserManagement.LAST_NAME_SWITCH, "newuser")],
    )
    def test_valid_user_update(client, server_config, pytestconfig, switch, value):
        with client:
[...]
            result = runner.invoke(
                user_update,
                args=[TestUserManagement.USER_TEXT, switch, value]
            )
            assert result.exit_code == 0
[...]
```
which should run the test six times, once for each switch.  (You could consider tossing in a case which specifies empty strings for the switch and value and see if that passes, too.)

(Hey, @dbutenhof, did you know that you can apply multiple `parametrize` decorators to the same function?  They run the test with the cross-product of all the combinations!  Pretty wild....)

Reply:
> (Hey, @dbutenhof, did you know that you can apply multiple `parametrize` decorators to the same function? They run the test with the cross-product of all the combinations! Pretty wild....)

I did not. I'm not entirely sure I'm happy you told me that. ;-)